dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -1372,6 +1383,11 @@ def _regenerate_histogram(
             new_bin_edges = np.linspace(
                 options["min_edge"], options["max_edge"], suggested_bin_count + 1
             )
+
+        # if it's not a float histogram, then assume it only contains integer values
+        if not is_float_histogram:
+            bin_edges = np.round(bin_edges)
```

Comment:
this is a big assumption ... can you comment on the rationale here?

Reply:
The same assumption was being made in `_assimilate_histogram` ([diff](https://github.com/capitalone/DataProfiler/pull/1071/files#diff-c46fb1aabfd8fc56a5d68b0ab6743d0c27918bc5b929538411a301472d116a43L1420)). I moved it from `_assimilate_histogram` to `_regenerate_histogram`.

We could also remove that rounding code entirely, but it causes a single test failure [here](https://github.com/junholee6a/DataProfiler/blob/assimilate_histogram_remove_dependency/dataprofiler/tests/profilers/test_text_column_profile.py#L285). I haven't been able to figure out what purpose the rounding fulfills here, as it existed since the first release. I kept it so that the behavior of `_regenerate_histogram` wouldn't change from before.

But you're right that it's a big assumption, and it's worth considering its removal. I'll open an alternative PR for that

Reply:
PR https://github.com/capitalone/DataProfiler/pull/1073 is an alternative to this PR where we remove the rounding code

Reply:
Okay, after extensive discussion with @taylorfturner @micdavis we have determined that this assumption is valid and all values passed into this function at the current placement in the workflow will all be numeric (either ints and floats). I think there should be an additional test where this line 1389 is called (or if a test already exists) to validate that bin edges are able to be rounded at that point in the workflow (i.e. is it possible to get a non-roundable value at this point in the code, which I believe is not possible).

Assuming this test is created I am good with this version of code and we can delete PR 1073

Reply:
I closed #1073

Reply:
@ksneab7 Thanks for the feedback! Line 1389 is called in the existing test `TestTextColumnProfiler.test_profile` (I've attached a screenshot of my debugger stopping at line 1389 while running the test, to demonstrate this). Also, if this rounding code is removed entirely, the output changes and this test fails, as I explained in https://github.com/capitalone/DataProfiler/pull/1073.

<img width="1470" alt="Screenshot 2024-01-09 at 2 31 52 PM" src="https://github.com/capitalone/DataProfiler/assets/53921230/217417f0-4899-4d1d-8707-8062eee619d6">

Does this fulfill what you meant, or did you have something else in mind?

Reply:
Nope this is exactly what I was looking for thank you!